src/assets/wise5/components/common/feedbackRule/FeedbackRuleEvaluator.ts

```diff
@@ -20,45 +20,92 @@ export class FeedbackRuleEvaluator {
     return this.getDefaultRule(this.component.getFeedbackRules());
   }
 
-  private satisfiesRule(response: CRaterResponse, feedbackRule: FeedbackRule): boolean {
+  private satisfiesRule(
+    response: CRaterResponse | CRaterResponse[],
+    feedbackRule: FeedbackRule
+  ): boolean {
     return this.isSpecialRule(feedbackRule)
       ? this.satisfiesSpecialRule(response, feedbackRule)
       : this.satisfiesSpecificRule(response, feedbackRule);
   }
 
-  private satisfiesSpecialRule(response: CRaterResponse, feedbackRule: FeedbackRule): boolean {
+  private satisfiesSpecialRule(
+    response: CRaterResponse | CRaterResponse[],
+    feedbackRule: FeedbackRule
+  ): boolean {
     return (
       this.satisfiesNonScorableRule(response, feedbackRule) ||
       this.satisfiesFinalSubmitRule(response, feedbackRule) ||
       this.satisfiesSecondToLastSubmitRule(response, feedbackRule)
     );
   }
 
-  private satisfiesFinalSubmitRule(response: CRaterResponse, feedbackRule: FeedbackRule): boolean {
-    return (
-      this.component.hasMaxSubmitCount() &&
-      this.component.hasMaxSubmitCountAndUsedAllSubmits(response.submitCounter) &&
-      FeedbackRule.isFinalSubmitRule(feedbackRule)
-    );
+  private satisfiesFinalSubmitRule(
+    response: CRaterResponse | CRaterResponse[],
+    feedbackRule: FeedbackRule
+  ): boolean {
+    if (response instanceof CRaterResponse) {
+      return (
+        this.hasMaxSubmitAndIsFinalSubmitRule(feedbackRule) &&
+        this.component.hasMaxSubmitCountAndUsedAllSubmits(response.submitCounter)
+      );
+    } else {
+      return (
+        this.hasMaxSubmitAndIsFinalSubmitRule(feedbackRule) &&
+        response.some((response: CRaterResponse) => {
+          return this.component.hasMaxSubmitCountAndUsedAllSubmits(response.submitCounter);
+        })
+      );
+    }
```

Comment:
Use ternary operator to reduce duplication and complexity
```suggestion
    return (
      this.hasMaxSubmitAndIsFinalSubmitRule(feedbackRule) &&
      (response instanceof CRaterResponse
        ? this.component.hasMaxSubmitCountAndUsedAllSubmits(response.submitCounter)
        : response.some((response: CRaterResponse) => {
            return this.component.hasMaxSubmitCountAndUsedAllSubmits(response.submitCounter);
          }))
    );
```